### DISCUSSION TITLE
How do I save measurement data to a specified location on my computer?

### DISCUSSION THREAD

**Discussion by vuhoang8** [tektronix/tm_devices#268] [Q&A] [answered]
I'm using tm_devices to program a Keithley 2612B SMU. So far, I've been able to conduct an I-V sweep and save the voltage and current data to two separate buffers that can be printed to the terminal. However, I am unsure of how to save the data stored in those buffers to a specific location on my PC. As an example, I am using the following code:

`smu.commands.savebuffer("smua.nvbuffer2", "CSV", "voltage_test001")`

After running this, I'm unable to locate where the file is. Is there another function I could use instead that allows me to specify the file's location?

**Comment by SydTenaglia** [comment, marked as answer]
The savebuffer() command saves the specified buffer to a USB flash drive inserted into the instrument front panel. The 2612B cannot access the file structure of a host PC, but you have a couple of options for work arounds:

You can save data to a USB drive that's plugged into the instrument and then transfer the data from the USB manually.
You can copy/paste the data printed to the terminal into excel
Or you can use the printbuffer(startIndex, endIndex, bufferVar, bufferVar2) command to return the buffer data to the pc and do some file manipulation within python:
```
def write_readings_to_file(file_path, floats):
    # This function writes the floating point data to the
    # target file.

    ofile = open(file_path, "a")  # Open/create the target data for DMM1
    tmp = 0
    for f in floats:
        if tmp % 2 == 0:    # indices divisible by 2 will hold the reading value; odd indices hold timestamp
            ofile.write("{0:.6e},".format(f))
        else:
            ofile.write("{0:.8e}\n".format(f))
        tmp += 1

    ofile.close()
    return
```
See the 2612B manual for more information about how the smu commands work: https://download.tek.com/manual/2600BS-901-01_C_Aug_2016_2.pdf
